frontend/src/lib/k8s/index.ts

```diff
@@ -135,6 +136,13 @@ export function useClustersConf(): ConfigState['clusters'] {
   return clusters;
 }
 
+export async function refreshConfig() {
+  const response = await post('/config/clusters/refresh', {}, false, {
+    timeout: 5 * 1000,
+  });
+  return response as ConfigState['clusters'];
```

Comment:
Probably should check if there's a 200 return code.

Reply:
The underlying `post` function which uses `request` takes care of status code checks and returns only `response.json()` as result.

Reply:
Opening this back up. 

> The underlying post function which uses request takes care of status code checks and returns only response.json() as result.
Not true. It still needs error handling.

Reply:
I have added a `try catch` block for handling errors but not sure how to check the response status code because the `request` function only returns `response.json()`. Can you point me to an example that I can refer to?

Reference:
https://github.com/kinvolk/headlamp/blob/d6df0869cd37af55f06f239219fcb4c1062517fd/frontend/src/lib/k8s/apiProxy.ts#L478-L487

https://github.com/kinvolk/headlamp/blob/d6df0869cd37af55f06f239219fcb4c1062517fd/frontend/src/lib/k8s/apiProxy.ts#L132

https://github.com/kinvolk/headlamp/blob/d6df0869cd37af55f06f239219fcb4c1062517fd/frontend/src/lib/k8s/apiProxy.ts#L218

Reply:
Yeah, the request checks the response.ok already. It's kind of an unfortunate API choice in some ways, since you can't check the response yourself. But it also means we don't need to handle all those cases ourselves. If you need to handle something other than 'ok' (200, 201 etc), then not using request() would be appropriate.

The request() returns `Promise.reject(error);` a few lines above the `return response.json()`.  So all not-ok responses should be handled by your try/catch.

The fetch API returns the response most of the time, except in exceptional circumstances (networks errors basically). https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API